### PR TITLE
Add search to docs site

### PR DESCRIPTION
**Pull request by elsom25**
Adding category search to our docs site.

## 🔍 Which search library?

- ❌ [tinysearch](https://github.com/tinysearch/tinysearch). WASM for speed, but search quality too low.
  - While it's blazingly fast, it's search result quality wasn't great with our fairly overlapping data sets. It's lack of [substring filtering](https://github.com/tinysearch/tinysearch?tab=readme-ov-file#limitations) proved problematic as well.
- ✅ [fuse.js](https://github.com/krisk/Fuse). No dependencies and JS only, with solid result responses.
  - Very actively maintained, has several levers for further tuning down the line, and has tooling we can opt to use to further improve performance in later cycles.

## 🗒️ What's left?

- [x] Generate search index
- [x] Add search to index.html (simple)
- [x] Make search results selection client-side
- [x] Improve html + css for a passable experience

## 👀 What results look like

image

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -13,6 +13,40 @@
...
+        await init('./public/tinysearch/engine_bg.wasm');
+      }
+      run();
+    </script>
+
+    <!-- Setup search functions -->
+    <script>
+      function categorySearch(query) {
+        const results = search(query, 5);
+        console.log(`Results: ${results}`);
```
Comment (Chris911) [inline docs/index.html:31]: Keep/remove?

File: dev/bin/generate_docs [hunk 2]
```diff
@@ -34,20 +53,48 @@ class GenerateDocs
...
-      attribute_data = JSON.load_file("#{Application.root}/dist/attributes.json")
+    def generate_search_index(verticals)
+      puts "Preparing search index"
+      puts "  → Parsing categories into search index"
+      search_index = verticals.flat_map do |vertical|
+        vertical['categories'].map do |category|
+          {
+            "title" => category.fetch("name"),
+            "url" => "?categoryId=#{CGI.escapeURIComponent(category.fetch("id"))}",
+            "body" => category.fetch("full_name"),
```
Comment (Chris911) [inline dev/bin/generate_docs:77]: Nit: the doc says `body` is optional. We could use `full_name` as the `title` and remove `body`.

**Review by Chris911 — COMMENTED** [review]
Code looks good but are the results ok? From the screenshot you shared + testing some of the sites listed in the tinysearch's README I'm getting less than ideal results.

File: dev/bin/generate_docs (hunk 2 above)
Comment (elsom25) [reply dev/bin/generate_docs:77]: Yeah I was torn on if search results with full name ended up being _too much_, but probably necessary given result quality

**Pull request merged**